src/0x.ts

```diff
@@ -311,6 +312,16 @@ export class ZeroEx {
                 }
             }, pollingIntervalMs);
         });
+
+        if (timeoutMs) {
+            return Promise.race([
```

Comment:
As far as I understand - this will reject but will not cancel the polling and we will continue to send requests

Reply:
That's a good point - adjusted the approach. Instead of using promise race, just set a "timeoutExceeded" bool to check with the interval function. If timeout has been exceeded, clear the interval and reject the promise.